pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManagerStatelessTest.java

```diff
@@ -1146,6 +1147,20 @@ public void testSegmentReplacementRegular()
     segmentLineage = SegmentLineageAccessHelper.getSegmentLineage(_propertyStore, OFFLINE_TABLE_NAME);
     assertNull(segmentLineage);
   }
+  @Test
+  public void testHandleEmptyServerTags()
+      throws Exception {
+    // Create an instance with no tags
+    Instance instance = new Instance("localhost", NUM_SERVER_INSTANCES , InstanceType.SERVER,
+        Collections.emptyList(), null, 0, 12345, 0, 0, false);
+
+    _helixResourceManager.addInstance(instance,false);
+    addFakeServerInstanceToAutoJoinHelixClusterWithEmptyTag(SERVER_NAME_UNTAGGED, false);
+
+    // Verify that the server is considered untagged
+    List<String> untaggedServers = _helixResourceManager.getOnlineUnTaggedServerInstanceList();
+    assertTrue(untaggedServers.contains(SERVER_NAME_UNTAGGED), "Server with empty tags should be considered untagged");
```

Comment:
Would be good to add the negative test as well - Servers with tag should not be considered untagged.

Reply:
Done.